File: 长度最小的子数组.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int left = 0;
        int n = nums.size();
        int sum = 0;
        int ans = n + 1;

        for (int right = 0; right < n; ++right) {
            sum += nums[right];
            while (sum - nums[left] >= target) {
                sum -= nums[left];
                left++;
            }
            if (sum >= target)
                ans = min(ans, right - left + 1);
        }
        return ans == n + 1 ? 0 : ans;
    }
};

File: 长度最小的子数组.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        ans, left,sum = len(nums) + 1, 0, 0
        for right, num in enumerate(nums):
            sum += num
            while sum - nums[left] >= target:
                sum -= nums[left]
                left += 1
            if sum >= target:
                ans = min(ans, right - left + 1)
        return ans if ans != len(nums) + 1 else 0
